### PR TITLE
Validate blocks

### PR DESCRIPTION
**Pull request by lukanus**

### CODE REVIEW

File: pkg/relay/submit.go [hunk 1]
```diff
@@ -31,7 +32,18 @@ func (rs *Relay) SubmitBlock(ctx context.Context, m *structs.MetricGroup, submit
 		"bid":       submitBlockRequest.Message.Value.String(),
 	})
 
-	_, err := verifyBlock(submitBlockRequest, rs.beaconState.Beacon())
+	resp, err := rs.bvc.ValidateBlock(ctx, &rpctypes.BuilderBlockValidationRequest{
+		BuilderSubmitBlockRequest: *submitBlockRequest,
+		RegisteredGasLimit:        10000000000,
```
Comment (aratz-lasa) [inline pkg/relay/submit.go:37]: Where do we get this value from?

**Review by aratz-lasa — COMMENTED** [review]
The PR looks really good! I just have questions / suggestions

File: cmd/dreamboat/main.go [hunk 2]
```diff
@@ -219,6 +224,24 @@ var flags = []cli.Flag{
 		Value:   2,
 		EnvVars: []string{"RELAY_SUBMISSION_LIMIT_BURST"},
 	},
+	&cli.StringFlag{
```
[the remark this answers is not in the document]
Comment (lukanus) [reply cmd/dreamboat/main.go:227]: I made it pretty simple, like without any preoptimisation. 
Assuming that:
* ipc is local anyway, so it wouldn't make this much sense. 
* I assumed that as http are separate network connections it would be under some kind of LB anyway. 
* ws acctually allows you to add more than one connection to round robin, but for simplicity I'm not reusing the same connection type, just fallback to the slower one (http). We can add this later but not sure if that should be the priority right now

File: pkg/client/sim/fallback/fallback.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	return len(f.clients)
+}
+
+func (f *Fallback) ValidateBlock(ctx context.Context, block *types.BuilderBlockValidationRequest) (rrr types.RpcRawResponse, err error) {
+	if len(f.clients) == 0 {
+		return rrr, client.ErrNotFound
+	}
+
+	for _, c := range f.clients {
+		rrr, err = c.ValidateBlock(ctx, block)
```
[the remark this answers is not in the document]
Comment (lukanus) [reply pkg/client/sim/fallback/fallback.go:37]: rrr is RpcRawResponse :P I can change that but after adding ipc that is not using jsonrpc I was wondering shouldn't I just return error from this function as json rpc isn't really used externally.

File: pkg/client/sim/transport/gethhttp/gethhttp.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+)
+
+type Client struct {
+	namespace string
+	client    *http.Client
+	address   string
+	l         log.Logger
+}
+
+func NewClient(address string, namespace string, l log.Logger) *Client {
```
[the remark this answers is not in the document]
Comment (lukanus) [reply pkg/client/sim/transport/gethhttp/gethhttp.go:21]: It's written by us, it's simpler, and was there before I knew we need to implement ipc ;D

File: pkg/client/sim/transport/gethws/gethws.go [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+				"response": r,
+			}).Error("error in async call")
+			continue
+		}
+		ch <- r
+	}
+}
+
+func (conn *Conn) readHandler() {
+
```
[the remark this answers is not in the document]
Comment (lukanus) [reply pkg/client/sim/transport/gethws/gethws.go:125]: The code was supposed to stop there and reflect upon it's existence ;>

File: pkg/client/sim/transport/gethws/gethws.go [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+func (conn *Conn) writeHandler() {
+	ticker := time.NewTicker(healthInterval)
+	defer conn.Close()
+
+	// allow to wait first full interval
+	conn.lastRead = time.Now()
+	for {
+		select {
+		case t := <-conn.input:
```
[the remark this answers is not in the document]
Comment (lukanus) [reply pkg/client/sim/transport/gethws/gethws.go:151]: https://google.github.io/styleguide/go/decisions#variable-names 
TLDR; If it's obvious and used "in place", I'm narrowing down var names to simple letters, as it's parsable better by (as least my) brain

File: pkg/client/sim/fallback/fallback.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (aratz-lasa) [reply pkg/client/sim/fallback/fallback.go]: Yeah, that sounds 👍

File: pkg/client/sim/transport/gethhttp/gethhttp.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (aratz-lasa) [reply pkg/client/sim/transport/gethhttp/gethhttp.go:22]: So is it redundant?

File: pkg/client/sim/transport/gethws/gethws.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (aratz-lasa) [reply pkg/client/sim/transport/gethws/gethws.go]: 😂😂😂

File: pkg/client/sim/transport/gethws/gethws.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (aratz-lasa) [reply pkg/client/sim/transport/gethws/gethws.go:154]: What about in? 😁

File: pkg/client/sim/transport/gethws/gethws.go [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+		ch <- r
+	}
+}
+
+func (conn *Conn) readHandler() {
+
+	// run parallel decoder
+	ch := make(chan []byte, 50)
+	defer close(ch)
+	for i := 0; i < 10; i++ {
```
[the remark this answers is not in the document]
Comment (lukanus) [reply pkg/client/sim/transport/gethws/gethws.go]: done

File: pkg/client/client.go [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package client
```
[the remark this answers is not in the document]
Comment (lukanus) [reply pkg/client/client.go:1]: done

**Review by aratz-lasa — APPROVED** [review]
LGTM
